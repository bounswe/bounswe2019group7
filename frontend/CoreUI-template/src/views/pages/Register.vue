<template>
  <div class="app flex-row align-items-center">
    <div class="container">
      <b-row class="justify-content-center">
        <b-col md="6" sm="8">
          <b-card no-body class="mx-4">
            <b-card-body class="p-4">
              <b-form>
                <h1>Register</h1>
                <p class="text-muted">Create your account</p>
                <b-input-group class="mb-3">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-user"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="text"
                    class="form-control"
                    placeholder="Name"
                    autocomplete="name"
                  />
                </b-input-group>

                <b-input-group class="mb-3">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-user"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="text"
                    class="form-control"
                    placeholder="Surname"
                    autocomplete="surname"
                  />
                </b-input-group>

                <b-input-group class="mb-3">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-user"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="text"
                    class="form-control"
                    placeholder="Username"
                    autocomplete="username"
                  />
                </b-input-group>

                <b-input-group class="mb-3">
                  <b-input-group-prepend>
                    <b-input-group-text>@</b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="text"
                    class="form-control"
                    placeholder="Email"
                    autocomplete="email"
                  />
                </b-input-group>

                <b-input-group class="mb-3">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-lock"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="password"
                    class="form-control"
                    placeholder="Password"
                    autocomplete="new-password"
                  />
                </b-input-group>

                <b-input-group class="mb-4">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-lock"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="password"
                    class="form-control"
                    placeholder="Repeat password"
                    autocomplete="new-password"
                  />
                </b-input-group>

                <b-input-group class="mb-4">
                  <b-input-group-prepend>
                    <b-input-group-text>
                      <i class="icon-location-pin"></i>
                    </b-input-group-text>
                  </b-input-group-prepend>
                  <b-form-input
                    type="text"
                    class="form-control"
                    placeholder="Location"
                    autocomplete="location"
                  />
                </b-input-group>

                <b-form-group label="User Type:" label-for="userType" :label-cols="3">
                  <b-form-radio-group
                    id="userType"
                    :plain="true"
                    :options="[
                      {text: 'Trader ',value: '1'},
                      {text: 'Basic User ',value: '2'}
                    ]"
                    checked="1"
                    stacked
                    v-on:change="changeUserType"
                  ></b-form-radio-group>
                </b-form-group>
                <div v-if="seen">
                  <b-input-group class="mb-4" id="creditCardDiv">
                    <b-input-group-prepend>
                      <b-input-group-text>
                        <i class="icon-credit-card"></i>
                      </b-input-group-text>
                    </b-input-group-prepend>
                    <b-form-input
                      type="text"
                      class="form-control"
                      placeholder="Credit Card Number(0000-0000-0000-0000)"
                      autocomplete="creditCardNumber"
                    />
                  </b-input-group>
                  <b-input-group class="mb-4" id="idNumberDiv">
                    <b-input-group-prepend>
                      <b-input-group-text>
                        <i class="fa fa-id-card-o fa-lg"></i>
                      </b-input-group-text>
                    </b-input-group-prepend>
                    <b-form-input
                      type="text"
                      class="form-control"
                      placeholder="Identity Number"
                      autocomplete="identityNumber"
                    />
                  </b-input-group>
                </div>

                <b-button variant="success" block>Create Account</b-button>
              </b-form>
            </b-card-body>
            <b-card-footer
              class="p-4"
              style="align-self:center; padding:1.5rem 10% !important; background-color:unset"
            >
              <b-row>
                <b-button size="lg" variant="google-plus" class="mr-1 btn-brand">
                  <i class="fa fa-google"></i>
                  <span>Sign up with Google</span>
                </b-button>
              </b-row>
            </b-card-footer>
          </b-card>
        </b-col>
      </b-row>
    </div>
  </div>
</template>

<script>
export default {
  name: "Register",
  data: function() {
    return {
      seen: true
    };
  },
  methods: {
    changeUserType() {
      console.log(this.seen);
      this.seen = !this.seen;
    }
  }
};
</script>
