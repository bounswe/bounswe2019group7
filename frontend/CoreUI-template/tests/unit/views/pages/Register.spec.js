import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Register from '@/views/pages/Register'

Vue.use(BootstrapVue)

describe('Register.vue', () => {
  it('has a name', () => {
    expect(Register.name).toMatch('Register')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Register', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.is(Register)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.find('h1').text()).toMatch('Register')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Register)
    expect(wrapper.element).toMatchSnapshot()
  })
})
