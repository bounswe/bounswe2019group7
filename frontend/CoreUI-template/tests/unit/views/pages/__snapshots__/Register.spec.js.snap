// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue renders correctly 1`] = `
<div
  class="app flex-row align-items-center"
>
  <div
    class="container"
  >
    <b-row-stub
      class="justify-content-center"
      tag="div"
    >
      <b-col-stub
        md="6"
        sm="8"
        tag="div"
      >
        <b-card-stub
          bodytag="div"
          class="mx-4"
          footertag="div"
          headertag="div"
          nobody="true"
          subtitle=""
          subtitletag="h6"
          subtitletextvariant="muted"
          tag="div"
          title=""
          titletag="h4"
        >
          <b-card-body-stub
            bodytag="div"
            class="p-4"
            subtitle=""
            subtitletag="h6"
            subtitletextvariant="muted"
            title=""
            titletag="h4"
          >
            <b-form-stub>
              <h1>
                Register
              </h1>
               
              <p
                class="text-muted"
              >
                Create your account
              </p>
               
              <b-input-group-stub
                class="mb-3"
                tag="div"
              >
                <b-input-group-prepend-stub
                  tag="div"
                >
                  <b-input-group-text-stub
                    tag="div"
                  >
                    <i
                      class="icon-user"
                    />
                  </b-input-group-text-stub>
                </b-input-group-prepend-stub>
                 
                <b-form-input-stub
                  autocomplete="username"
                  class="form-control"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </b-input-group-stub>
               
              <b-input-group-stub
                class="mb-3"
                tag="div"
              >
                <b-input-group-prepend-stub
                  tag="div"
                >
                  <b-input-group-text-stub
                    tag="div"
                  >
                    @
                  </b-input-group-text-stub>
                </b-input-group-prepend-stub>
                 
                <b-form-input-stub
                  autocomplete="email"
                  class="form-control"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </b-input-group-stub>
               
              <b-input-group-stub
                class="mb-3"
                tag="div"
              >
                <b-input-group-prepend-stub
                  tag="div"
                >
                  <b-input-group-text-stub
                    tag="div"
                  >
                    <i
                      class="icon-lock"
                    />
                  </b-input-group-text-stub>
                </b-input-group-prepend-stub>
                 
                <b-form-input-stub
                  autocomplete="new-password"
                  class="form-control"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </b-input-group-stub>
               
              <b-input-group-stub
                class="mb-4"
                tag="div"
              >
                <b-input-group-prepend-stub
                  tag="div"
                >
                  <b-input-group-text-stub
                    tag="div"
                  >
                    <i
                      class="icon-lock"
                    />
                  </b-input-group-text-stub>
                </b-input-group-prepend-stub>
                 
                <b-form-input-stub
                  autocomplete="new-password"
                  class="form-control"
                  placeholder="Repeat password"
                  type="password"
                  value=""
                />
              </b-input-group-stub>
               
              <b-button-stub
                block="true"
                event="click"
                routertag="a"
                tag="button"
                target="_self"
                type="button"
                variant="success"
              >
                Create Account
              </b-button-stub>
            </b-form-stub>
          </b-card-body-stub>
           
          <b-card-footer-stub
            class="p-4"
            footertag="div"
          >
            <b-row-stub
              tag="div"
            >
              <b-col-stub
                cols="6"
                tag="div"
              >
                <b-button-stub
                  block="true"
                  class="btn btn-facebook"
                  event="click"
                  routertag="a"
                  tag="button"
                  target="_self"
                  type="button"
                  variant="secondary"
                >
                  <span>
                    facebook
                  </span>
                </b-button-stub>
              </b-col-stub>
               
              <b-col-stub
                cols="6"
                tag="div"
              >
                <b-button-stub
                  block="true"
                  class="btn btn-twitter"
                  event="click"
                  routertag="a"
                  tag="button"
                  target="_self"
                  type="button"
                  variant="secondary"
                >
                  <span>
                    twitter
                  </span>
                </b-button-stub>
              </b-col-stub>
            </b-row-stub>
          </b-card-footer-stub>
        </b-card-stub>
      </b-col-stub>
    </b-row-stub>
  </div>
</div>
`;
